xml/fingerprints.xsd

```diff
@@ -7,6 +7,9 @@
       <xsd:element name="fingerprint" type="fingerprint_element" minOccurs="1" maxOccurs="unbounded"/>
     </xsd:sequence>
     <xsd:attribute name="matches" type="xsd:string" use="optional"/>
+    <xsd:attribute name="protocol" type="xsd:string" use="optional"/>
+    <xsd:attribute name="type" type="xsd:string" use="optional"/>
+    <xsd:attribute name="priority" type="xsd:string" use="optional"/>
```

Comment:
May need to talk this one through a bit.  The values seem a little arbitrary and maybe only specific to your current use case.